ui/src/app/kvm/components/KVMActionFormWrapper/ComposeForm/StorageTable/PoolSelect/PoolSelect.tsx

```diff
@@ -0,0 +1,179 @@
+import { useFormikContext } from "formik";
+import React from "react";
+import { useSelector } from "react-redux";
+import { useParams } from "react-router";
+
+import type { ComposeFormValues, DiskField } from "../../ComposeForm";
+import type { PodDetails } from "app/store/pod/types";
+import type { RootState } from "app/store/root/types";
+import podSelectors from "app/store/pod/selectors";
+import { formatBytes } from "app/utils";
+import ContextualMenu from "app/base/components/ContextualMenu";
+import Meter from "app/base/components/Meter";
+
+type Props = {
+  disk: DiskField;
+  selectPool: (poolName?: string) => void;
+};
+
+export const PoolSelect = ({ disk, selectPool }: Props): JSX.Element => {
+  const { id } = useParams();
+  const pod = useSelector((state: RootState) =>
+    podSelectors.getById(state, Number(id))
+  ) as PodDetails;
+  const { values } = useFormikContext<ComposeFormValues>();
+
+  const { disks } = values;
+  const requests: { [location: string]: number } = disks.reduce(
+    (requests, disk) => {
+      if (requests[disk.location]) {
+        requests[disk.location] += disk.size;
+      } else {
+        requests[disk.location] = disk.size;
+      }
+      return requests;
+    },
+    {}
+  );
+  const sortedPools = [
+    pod.storage_pools.find((pool) => pool.id === pod.default_storage_pool),
+    ...pod.storage_pools.filter((pool) => pool.id !== pod.default_storage_pool),
+  ].filter(Boolean);
+
+  return (
+    <ContextualMenu
+      className="kvm-pool-select"
+      constrainPanelWidth
+      dropdownClassName="kvm-pool-select__dropdown"
+      dropdownContent={
```

Comment:
possible to move this to a function so the rest of the `ContextualMenu` props aren't buried at the bottom of the large block?

Reply:
Absolutely!